crates/hotshot/types/src/epoch_membership.rs

```diff
@@ -323,6 +287,46 @@ where
             // Remove the epoch from the catchup map to indicate that the catchup is complete
             self.catchup_map.lock().await.remove(&current_fetch_epoch);
         }
+
+        let root_leaf = match self.fetch_stake_table(epoch).await {
+            Ok(root_leaf) => root_leaf,
+            Err(err) => {
+                fetch_epochs.push((epoch, epoch_tx));
+                self.catchup_cleanup(epoch, fetch_epochs, err).await;
+                return;
+            },
+        };
```

Comment:
If we're doing this here it's just for getting the root leaf and we only need it if we fail to fetch, so lets move this to the Err block below